usage_analytics/analytics_http_client.py

```diff
@@ -0,0 +1,39 @@
+import httpx
+from .analytics_definition import ANALYTICS_TO_TRACK
+
+
+class AnalyticsHttpClient(httpx.Client):
+    """
+    Custom HTTP client that automatically adds the calling class to all requests.
+    """
+
+    def __init__(self, module_name: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.module_name = module_name
+
+    def request(self, method, url, *args, **kwargs):
+        headers = kwargs.get("headers", {})
+        if headers is None:
+            headers = {}
+
+        kwargs["headers"] = self._add_analytics_headers(headers)
```

Comment:
This patterns looks a bit smelly.  

The kwargs are a dict and thus mutable.  
Modifying them might change values somewhere up in the callstack and thus introduce side-effects.

The clean way is to construct a new kwargs element that references all the existing ones, but changes the new ones.

I think this should work

```python
new_kwargs = {
  **kwargs,
  "headers": self._add_analytics_headers(headers)
}
```

I am not sure how exactly kwargs work in python, but this type of pattern is a huge issue of passing pointers.